pkg/flavor/flavor_test.go

```diff
@@ -109,31 +145,76 @@ var _ = Describe("Flavor and Preferences", func() {
 
 				f, err := flavorMethods.FindFlavorSpec(vm)
 				Expect(err).ToNot(HaveOccurred())
-				Expect(*f).To(Equal(flavor.Spec))
+				Expect(*f).To(Equal(clusterFlavor.Spec))
 			})
 
-			It("fails when flavor does not exist", func() {
+			It("find fails when flavor does not exist", func() {
 				vm.Spec.Flavor.Name = "non-existing-flavor"
-
 				_, err := flavorMethods.FindFlavorSpec(vm)
 				Expect(err).To(HaveOccurred())
 				Expect(errors.IsNotFound(err)).To(BeTrue())
 			})
+
+			It("store VirtualMachineClusterFlavor ControllerRevision", func() {
+
+				clusterFlavorControllerRevision, err := flavor.CreateFlavorControllerRevision(vm, flavor.GetRevisionName(vm.Name, clusterFlavor.Name, clusterFlavor.UID, clusterFlavor.Generation), &clusterFlavor.Spec)
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedRevisionNamePatch, err := flavor.GenerateRevisionNamePatch(clusterFlavorControllerRevision, nil)
+				Expect(err).ToNot(HaveOccurred())
+
+				vmInterface.EXPECT().Patch(vm.Name, types.JSONPatchType, expectedRevisionNamePatch, &metav1.PatchOptions{})
+
+				expectControllerRevisionCreation(clusterFlavorControllerRevision)
+
+				err = flavorMethods.StoreControllerRevisions(vm)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(vm.Spec.Flavor.RevisionName).To(Equal(clusterFlavorControllerRevision.Name))
+
+			})
+
+			It("store returns a nil revision when RevisionName already populated", func() {
+				clusterFlavorControllerRevision, err := flavor.CreateFlavorControllerRevision(vm, flavor.GetRevisionName(vm.Name, clusterFlavor.Name, clusterFlavor.UID, clusterFlavor.Generation), &clusterFlavor.Spec)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = virtClient.AppsV1().ControllerRevisions(vm.Namespace).Create(context.Background(), clusterFlavorControllerRevision, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				vm.Spec.Flavor = &v1.FlavorMatcher{
+					Name:         clusterFlavor.Name,
+					RevisionName: clusterFlavorControllerRevision.Name,
+					Kind:         apiflavor.ClusterSingularResourceName,
+				}
+				err = flavorMethods.StoreControllerRevisions(vm)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(vm.Spec.Flavor.RevisionName).To(Equal(clusterFlavorControllerRevision.Name))
+			})
+
+			It("store fails when flavor does not exist", func() {
+				vm.Spec.Flavor.Name = "non-existing-flavor"
+
+				err := flavorMethods.StoreControllerRevisions(vm)
+				Expect(err).To(HaveOccurred())
+				Expect(errors.IsNotFound(err)).To(BeTrue())
+			})
 		})
 
 		Context("Using namespaced Flavor", func() {
-			var flavor *flavorv1alpha1.VirtualMachineFlavor
+			var f *flavorv1alpha1.VirtualMachineFlavor
```

Comment:
Can you keep the old name?

Reply:
I think I did this because of the import of "kubevirt.io/kubevirt/pkg/flavor" above. I should likely alias that to something like flavorpkg?